### PR TITLE
CM-259: Encrypting & Decrypting a String in C#

### PR DESCRIPTION
**Pull request by dgarcia202**

### CODE REVIEW

File: authorization-dotnet/EncryptingAndDecryptingStringInCSharp/EncryptingAndDecryptingStringInCSharp/StringEncryptionService.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    public class StringEncryptionService
+    {
+        private byte[] IV =
+            {
+                0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+                0x09, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16
+            };
+
+        private byte[] DeriveKeyFromPassword(string password) =>
+            Rfc2898DeriveBytes.Pbkdf2(Encoding.Unicode.GetBytes(password), Array.Empty<byte>(), 1000, HashAlgorithmName.SHA384, 16);
```
Comment (mahrncic) [inline authorization-dotnet/EncryptingAndDecryptingStringInCSharp/EncryptingAndDecryptingStringInCSharp/StringEncryptionService.cs:15]: Please create "explanatory" variables for the `Pbkdf2()` method parameters. It is unclear what `Array.Empty<byte>()`, `1000`, `HashAlgorithmName.SHA384` and `16` values represent. So, can you convert your 
`DeriveKeyFromPassword()` method into a multi-line one (add curly braces) and initialize these variables with meaningful names, which will then be passed into the `Pbkdf2()` method? The code will look a lot cleaner, thanks!

File: authorization-dotnet/EncryptingAndDecryptingStringInCSharp/Tests/StringEncryptionTests.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            const string passphrase = "myPassword";
+            const string clearText = "We use encryption to obscure a piece of information.";
+
+            var encrypted = await encryptionService.EncryptAsync(clearText, passphrase);
+            var decrypted = await encryptionService.DecryptAsync(encrypted, passphrase);
+
+            Assert.Equal(clearText, decrypted);
+        }
+    }
+}
```
Comment (mahrncic) [inline authorization-dotnet/EncryptingAndDecryptingStringInCSharp/Tests/StringEncryptionTests.cs:20]: Can we please add one test for a negative scenario? For example, when using the invalid passphrase for decryption, the new test method could be called something like `WhenDecryptingAnEncryptedTextWithInvalidPassPhrase_ThenOriginalTextIsWrong`.

**Review by mahrncic — CHANGES_REQUESTED** [review]
It looks good, two small changes required.

File: authorization-dotnet/EncryptingAndDecryptingStringInCSharp/EncryptingAndDecryptingStringInCSharp/StringEncryptionService.cs (hunk 1 above)
Comment (dgarcia202) [reply authorization-dotnet/EncryptingAndDecryptingStringInCSharp/EncryptingAndDecryptingStringInCSharp/StringEncryptionService.cs]: fixed

File: authorization-dotnet/EncryptingAndDecryptingStringInCSharp/Tests/StringEncryptionTests.cs (hunk 2 above)
Comment (dgarcia202) [reply authorization-dotnet/EncryptingAndDecryptingStringInCSharp/Tests/StringEncryptionTests.cs]: fixed

**Review by mahrncic — APPROVED** [review]

File: authorization-dotnet/EncryptingAndDecryptingStringInCSharp/EncryptingAndDecryptingStringInCSharp/StringEncryptionService.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace EncryptingAndDecryptingStringInCSharp
+{
+    public class StringEncryptionService
+    {
+        private byte[] IV =
+            {
+                0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+                0x09, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16
+            };
```
Comment (mahrncic) [inline authorization-dotnet/EncryptingAndDecryptingStringInCSharp/EncryptingAndDecryptingStringInCSharp/StringEncryptionService.cs:12]: Please fix the indentation, curly braces should be right below the letter "p" of the `private` keyword, same as you did for the methods in this class :)

**Review by mahrncic — CHANGES_REQUESTED** [review]
Just one small change from the senior editor review, please also modify the article code snippet related to this indentation change

File: authorization-dotnet/EncryptingAndDecryptingStringInCSharp/EncryptingAndDecryptingStringInCSharp/StringEncryptionService.cs (hunk 3 above)
Comment (dgarcia202) [reply authorization-dotnet/EncryptingAndDecryptingStringInCSharp/EncryptingAndDecryptingStringInCSharp/StringEncryptionService.cs]: fixed
